type_classes.md

```diff
@@ -106,12 +227,19 @@ class Inhabited (a : Type u) where
 -- Inhabited.default : {a : Type u} → [self : Inhabited a] → a
 # end Ex
 ```
-Note `Inhabited.default` doesn't have any explicit argument.
+Note `Inhabited.default` doesn't have any explicit argument, it only has an implicit Type.
+In other words it is not a function, it is an object.
 
 An element of the class ``Inhabited a`` is simply an expression of the form ``Inhabited.mk x``, for some element ``x : a``.
 The projection ``Inhabited.default`` will allow us to "extract" such an element of ``a`` from an element of ``Inhabited a``.
 Now we populate the class with some instances:
 
+BUGBUG: what does "element" mean?  Is this a synonym for "instance"?
```

Comment:
`x` is an element of `a` is just text for `x : a`. Since a class is just a type as well saying that we have an element of the class `Inhabited a` is technically valid for saying that we have an instance of `Inhabited a` since instances `Inhabited a` are just values of type `Inhabited a`.

I think the best solution would be to use instance instead of element when referring to elements of a class and element when referring to "regular" types. If required one could also explain the idea that `class = structure + type class resolution` above.